setup.py

```diff
@@ -15,6 +15,19 @@
 import setuptools
 from setuptools import find_packages, setup
 
+print(
+    """
+===============================DEBUG ASSISTANCE===============================
+If you are seeing a compilation error below this message please try the
+following steps to successfully install cryptography:
+1) Upgrade to the latest pip and try again. This will fix errors for most
+   users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
+2) Read https://cryptography.io/en/latest/installation.html for specific
+   instructions for your platform.
+===============================DEBUG ASSISTANCE===============================
+"""
+)
```

Comment:
Should we only print this if an exception is raised?

Reply:
If there's a way to do that without messing with pip's own behaviors sure. I guess we could catch and rethrow?